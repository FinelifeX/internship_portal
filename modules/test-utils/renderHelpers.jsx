/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import TR from 'react-test-renderer';
import { render as RTLRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

/**
 * Render React component
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 * @returns {JSX.Element}
 */
export const render = (Component, { defaultProps, props, children } = {}) => (
  <Component {...defaultProps} {...props}>
    {children}
  </Component>
);

/**
 * Render React component inside MemoryRouter
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 * @param routerOptions
 * @returns {JSX.Element}
 */
export const renderWithRouter = (
  Component,
  { defaultProps, props, children } = {},
  routerOptions = {},
) => (
  <MemoryRouter {...routerOptions}>
    <Component {...defaultProps} {...props}>
      {children}
    </Component>
  </MemoryRouter>
);

/**
 * Shallow render component using "react-test-renderer"
 */
export const shallow = TR.create;

/**
 * Render component using "render()" from "@testing-library/react"
 * @type {import('@testing-library/react').render}
 */
export const mount = RTLRender;
