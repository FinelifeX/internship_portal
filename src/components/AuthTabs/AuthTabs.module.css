.nav {
  border-bottom: 1px solid rgba(0, 0, 0, 0.06);
  margin-bottom: 22px;
}

.navList {
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  margin: 0;
  padding: 0;
}

.navItem:not(:last-child) {
  margin-right: 32px;
}

.navItemLink {
  display: block;
  padding-bottom: 12px;
  color: rgba(0, 0, 0, 0.85);
}

.navItemLinkActive {
  border-bottom: 2px solid #1890ff;
  color: #1890ff;
}
