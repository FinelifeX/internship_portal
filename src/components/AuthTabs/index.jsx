import React from 'react';
import { NavLink } from 'react-router-dom';
import { AuthRoutes } from '../../routes/auth';
import styles from './AuthTabs.module.css';

const AuthTabs = () => (
  <nav className={styles.nav}>
    <ul className={styles.navList}>
      <li className={styles.navItem}>
        <NavLink
          to={AuthRoutes.Login}
          exact
          className={styles.navItemLink}
          activeClassName={styles.navItemLinkActive}
        >
          Login
        </NavLink>
      </li>
      <li className={styles.navItem}>
        <NavLink
          to={AuthRoutes.SignUp}
          exact
          className={styles.navItemLink}
          activeClassName={styles.navItemLinkActive}
        >
          Sign Up
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default AuthTabs;
