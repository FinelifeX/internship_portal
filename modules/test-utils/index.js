import * as RenderHelpers from './renderHelpers';

export { RenderHelpers };
