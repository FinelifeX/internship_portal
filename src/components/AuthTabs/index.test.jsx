import { shallow, mount, renderWithRouter } from 'test-utils/renderHelpers';
import AuthTabs from './index';

describe('AuthTabs component', () => {
  const TEST_ID_LOGIN = 'login';
  const TEST_ID_SIGNUP = 'signup';

  it('should render correctly', () => {
    const tree = shallow(renderWithRouter(AuthTabs)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  const renderComponent = () => mount(renderWithRouter(AuthTabs));

  it('should have links to login and sign-up pages', () => {
    const { getByTestId } = renderComponent();
    const loginLink = getByTestId(TEST_ID_LOGIN);
    const signUpLink = getByTestId(TEST_ID_SIGNUP);

    expect(loginLink).toBeInTheDocument();
    expect(signUpLink).toBeInTheDocument();
  });
});
