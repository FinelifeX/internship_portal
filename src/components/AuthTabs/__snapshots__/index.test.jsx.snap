// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthTabs component should render correctly 1`] = `
<nav
  className="nav"
>
  <ul
    className="navList"
  >
    <li
      className="navItem"
    >
      <a
        aria-current={null}
        className="navItemLink"
        data-testid="login"
        href="/auth/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="navItem"
    >
      <a
        aria-current={null}
        className="navItemLink"
        data-testid="signup"
        href="/auth/sign-up"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </li>
  </ul>
</nav>
`;
